openedx/features/edly/api/serializers.py

```diff
@@ -1,29 +1,45 @@
+"""
+Serializers for edly_api
+"""
+import json
+
 from rest_framework import serializers
 
 from openedx.core.djangoapps.site_configuration.helpers import get_value_for_org
 
 
 class UserSiteSerializer(serializers.Serializer):
+    """
+    Serializer for user_sites endpoint
+    """
     app_config = serializers.SerializerMethodField()
     site_data = serializers.SerializerMethodField()
+    mobile_enabled = serializers.SerializerMethodField()
 
     def get_app_config(self, obj):
-        mobile_app_config = get_value_for_org(
-            self.context['edly_sub_org_of_user'].edx_organization.short_name,
-            'MOBILE_APP_CONFIG',
-            default={}
-        )
-        url = get_value_for_org(
-            self.context['edly_sub_org_of_user'].edx_organization.short_name,
-            'SITE_NAME',
-            default=''
-        )
-        protocol = 'https' if self.context['request'].is_secure() else 'http'
-        mobile_app_config['API_HOST_URL'] = '{}://{}'.format(protocol, url) if url else ''
-        mobile_app_config['ORGANIZATION_CODE'] = self.context['edly_sub_org_of_user'].edx_organization.short_name
-        return mobile_app_config
+        """
+        Returns mobile specific data of a site using site configuration
+        """
+        if self.get_mobile_enabled(obj):
+            mobile_app_config = get_value_for_org(
+                self.context['edly_sub_org_of_user'].edx_organization.short_name,
+                'MOBILE_APP_CONFIG',
+                default={}
+            )
+            url = get_value_for_org(
+                self.context['edly_sub_org_of_user'].edx_organization.short_name,
+                'SITE_NAME',
+                default=''
+            )
+            protocol = 'https' if self.context['request'].is_secure() else 'http'
+            mobile_app_config['API_HOST_URL'] = '{}://{}'.format(protocol, url) if url else ''
+            mobile_app_config['ORGANIZATION_CODE'] = self.context['edly_sub_org_of_user'].edx_organization.short_name
+            return str(json.dumps(mobile_app_config))
```

Comment:
`json.dump` should already return a string. Why typecasting again?

Reply:
@rehan99000 That was the requirement of the mobile team. They wanted JSON data wrapped in another string.
```
"app_config": "{\"COURSE_SHARING_ENABLED\": true, \"COURSE_VIDEOS_ENABLED\": false, \"COURSE_DATES_ENABLED\": true, \"DISCUSSIONS_ENABLED\": false, \"API_HOST_URL\": \"http://edx.devstack.lms:18000\", \"ORGANIZATION_CODE\": \"edx\"}",
```